src/main/webapp/app/account/settings/settings.component.ts

```diff
@@ -1,20 +1,33 @@
 import { Component, OnInit, OnChanges } from '@angular/core';
-
-import { Principal, AccountService } from '../../shared';
+import {Principal, AccountService, Account} from '../../shared';
+import {PasswordService} from '../password/password.service';
+import {LoginService} from '../../shared/login/login.service';
 
 @Component({
     selector: 'jhi-settings',
     templateUrl: './settings.component.html'
 })
 export class SettingsComponent implements OnInit, OnChanges {
+    fieldValues: { [key: string]: string; } = {};
     error: string;
     success: string;
     settingsAccount: any;
     languages: any[];
     email: string;
+    doNotMatch: string;
+    passwordError: string;
+    passwordSuccess: string;
+    account: Account;
+    password: string;
+    currentPassword: string;
+    confirmPassword: string;
+    wrongPassword: boolean;
+
     constructor(
-        private account: AccountService,
-        private principal: Principal
+        private accountService: AccountService,
+        private principal: Principal,
```

Comment:
What's a principal?

Reply:
I don't know, it was already in settings.component. Looks like its maybe used to identify the user.